src/js/fluxcontroller.js

```diff
@@ -34,14 +34,53 @@ define(function (require, exports, module) {
 
     var locks = require("./locks"),
         events = require("./events"),
-        storeIndex = require("./stores/index"),
-        actionIndex = require("./actions/index"),
         AsyncDependencyQueue = require("./util/async-dependency-queue"),
         synchronization = require("./util/synchronization"),
         performance = require("./util/performance"),
         log = require("./util/log");
 
     /**
+     * Automatically loads all js files in ./actions folder
+     * replacing their paths with dot separated keys
+     *
+     * @param {function} requireContext require.context call of Webpack pointing to the actions folder
+     *
+     * @return {function()} when called, returns all action modules loaded
+     */
+    var requireActions = function (requireContext) {
+        return requireContext.keys().reduce(function (actionMap, actionKey) {
+            var actionId = actionKey.substring(2, actionKey.indexOf(".js")).replace(/\//gi, ".");
+
+            actionMap[actionId] = requireContext(actionKey);
+            return actionMap;
+        }, {});
+    };
+
+    /**
+     * Automatically loads all js files in ./stores folder
+     *
+     * @param {function} requireContext require.context call of Webpack pointing to stores folder
+     *
+     * @return {function()} when called, will initialize all stores
+     */
+    var requireStores = function (requireContext) {
+        return requireContext.keys().reduce(function (storeMap, storeKey) {
+            var storeId = storeKey.substring(2, storeKey.indexOf(".js")),
+                Store = requireContext(storeKey);
+
+            storeMap[storeId] = new Store();
+            return storeMap;
+        }.bind(this), {});
+    };
+
+    /**
+     * Raw (unsynchronized) action modules indiced by dot separated path as IDs
+     *
+     * @type {objects}
+     */
+    var actionIndex = requireActions(require.context("./actions", true, /^\.\/.*\.js$/));
```

Comment:
What do you think about moving the context creation code into the `requireActions` function?